File: week3/Eren-Taskiran/2-add-two-numbers.java
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummy=new ListNode();
        ListNode curr=dummy;
        ListNode temp1=l1;
        ListNode temp2=l2;
        int carry=0;
        while(temp1!=null ||temp2!=null){
            int sum=carry;
            if(temp1!=null){
                sum+=temp1.val;
            }
             if(temp2!=null){
                sum+=temp2.val;
            }
            ListNode newNode=new ListNode(sum%10);
            carry=sum/10;
            curr.next=newNode;
            curr=curr.next;
            if(temp1!=null){
                temp1=temp1.next;
            }
             if(temp2!=null){
                temp2=temp2.next;
            }
        }
        if(carry>0){
            ListNode newNode=new ListNode(carry);
            curr.next=newNode;
        }
        return dummy.next;

        
    }
}

File: week3/atakantekatan/2. Add Two Numbers.py
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        ans = None
        d1, d2 = 1, 1
        n1, n2 = 0, 0
        while l1 or l2:
            if l1 != None:
                n1 += d1*l1.val
                d1 *= 10
                l1 = l1.next
            if l2 != None:
                n2 += d2*l2.val
                d2 *= 10
                l2 = l2.next
        SUM = n1 + n2
        for digit in str(SUM):
            new = ListNode(int(digit))
            if ans is None:
                ans = new
            else:
                new.next = ans
                ans = new
        return ans